### PR TITLE
remove undefined props from getServerSideProps to be able to JSON.stringify them

### PR DESCRIPTION
**Pull request by jamesgorrie**
As next needs to serialise `getServerSideProps`, you can't have undefined values in there.

error

### CODE REVIEW

**Review by davidpmccormick — APPROVED** [review]

File: catalogue/webapp/pages/images.tsx [hunk 1]
```diff
@@ -28,13 +28,13 @@ import SearchForm from '@weco/common/views/components/SearchForm/SearchForm';
...
 import useSavedSearchState from '@weco/common/hooks/useSavedSearchState';
 import useHotjar from '@weco/common/hooks/useHotjar';
-import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';
 import SearchTabs from '@weco/common/views/components/SearchTabs/SearchTabs';
 import SearchNoResults from '../components/SearchNoResults/SearchNoResults';
 import {
   getGlobalContextData,
   WithGlobalContextData,
 } from '@weco/common/views/components/GlobalContextProvider/GlobalContextProvider';
+import { removeUndefinedProps } from '@weco/common/utils/json';
```
Comment (davidpmccormick) [inline catalogue/webapp/pages/images.tsx:37]: needs a FlowFixMe I think

File: catalogue/webapp/pages/progress.tsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import {
   getGlobalContextData,
   WithGlobalContextData,
 } from '@weco/common/views/components/GlobalContextProvider/GlobalContextProvider';
+import { removeUndefinedProps } from '@weco/common/utils/json';
```
Comment (davidpmccormick) [inline catalogue/webapp/pages/progress.tsx:12]: FlowFixMe

File: catalogue/webapp/pages/work.tsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ import {
   getGlobalContextData,
   WithGlobalContextData,
 } from '@weco/common/views/components/GlobalContextProvider/GlobalContextProvider';
+import { removeUndefinedProps } from '@weco/common/utils/json';
```
Comment (davidpmccormick) [inline catalogue/webapp/pages/work.tsx:13]: FlowFixMe

File: catalogue/webapp/pages/progress.tsx (hunk 2 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/progress.tsx:12]: No need, ts.

File: catalogue/webapp/pages/work.tsx (hunk 3 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/work.tsx:13]: This is TS file.

File: catalogue/webapp/pages/images.tsx (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/images.tsx:37]: Nope, these are TS files.

**Comment by jamesgorrie** [comment]
@davidpmccormick the flowfixme was needed [here](https://github.com/wellcomecollection/wellcomecollection.org/pull/5789/commits/7688266e6895b91a38e903baef601a0766f6cc7b)

**Review by ishouty — APPROVED** [review]

**Pull request merged**